### PR TITLE
Added shipper test that validates if shipper supports tsdb snapshots.

### PR DESCRIPTION
**Pull request by Bplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: pkg/shipper/shipper_test.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	storage := newInMemStorage(t)
+	shipper := New(
+		logger,
+		nil,
+		dir,
+		storage,
+		IsULIDDir,
+	)
+
+	ctx, cancel := context.WithCancel(context.TODO())
```
Comment (fabxc) [inline pkg/shipper/shipper_test.go]: `context.Background()` is fine here. Per docs, TODO is only for places where you want to connect it later with an incoming context.

File: pkg/shipper/shipper_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	defer cancel()
+
+	// Start shipping blocks.
+	go func() { testutil.Ok(t, shipper.Run(ctx)) }()
+
+	baseT := timestamp.FromTime(time.Now()) / 1000 * 1000
```
Comment (fabxc) [inline pkg/shipper/shipper_test.go]: I think we don't need the second alignment for this test.

File: pkg/shipper/shipper_test.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		baseT := baseT + (i * 300)
+		baseV := float64(i * 3)
+
+		a.Add(labels.FromStrings("a", labelVal), baseT+100, baseV+1)
+		a.Add(labels.FromStrings("a", labelVal), baseT+200, baseV+2)
+		a.Add(labels.FromStrings("a", labelVal), baseT+300, baseV+3)
+		testutil.Ok(t, a.Commit())
+
+		testutil.Ok(t, p.Snapshot(dir))
+		waitForShipment(t, ctx, storage, len(p.Blocks()))
```
Comment (fabxc) [inline pkg/shipper/shipper_test.go]: So `Snapshot()` creates a new sub-directory for each call to snapshot. The snapshotted blocks will also be overlapping in general. The shipper watches the normal data dir though and checks for top-level items that are ULID dirs.

So this should not work I think.

File: pkg/shipper/shipper_test.go (hunk 1 above)
Comment (Bplotka) [reply pkg/shipper/shipper_test.go]: when I was list visiting it... there was `...and tests` comment. (: Either my memory is weak or they changed that recently.

File: pkg/shipper/shipper_test.go (hunk 3 above)
Comment (fabxc) [reply pkg/shipper/shipper_test.go]: Ah, you are snapshotting directly back into the main dir. Was confusing that with where we push snapshots in Prometheus.
I thought though that we even added a safety guard forbidding that call with the dir itself.

Comment (fabxc) [reply pkg/shipper/shipper_test.go]: Mhh https://github.com/prometheus/tsdb/blob/master/db.go#L584

Comment (Bplotka) [reply pkg/shipper/shipper_test.go]: It is forbidden, I have special dir just for this. (:

File: pkg/shipper/shipper_test.go (hunk 1 above)
Comment (fabxc) [reply pkg/shipper/shipper_test.go]: Doesn't seem so anymore: https://godoc.org/context#TODO

**Comment by fabxc** [comment]
Aside from other comments looks good 👍

**Pull request merged**